File: page_objects/pages/sign_in.py
from playwright.sync_api import Locator

from page_objects.layouts.main import MainLayout


class SignInPage(MainLayout):
    @property
    def url(self) -> str:
        return f"{self._global_settings.frontend_base_url}/sign-in"

    @property
    def email_input(self) -> Locator:
        return self._page.locator("[data-test-id='email-input']")

    @property
    def password_input(self) -> Locator:
        return self._page.locator("[data-test-id='password-input']")

    @property
    def forgot_password_link(self) -> Locator:
        return self._page.locator("[data-test-id='forgot-password-link']")

    @property
    def sign_in_button(self) -> Locator:
        return self._page.locator("[data-test-id='login-button']")

    @property
    def sign_in_error_alert(self) -> Locator:
        return self._page.locator("[data-test-id='sign-in-error-alert']")

    def navigate(self) -> None:
        self._page.goto(url=self.url, wait_until="load")

File: _TYPESCRIPT/src/page-objects/frontend/pages/sign-in.ts
import { DefaultLayout } from "@page-objects/frontend/layouts/default";
import type { Locator } from "@playwright/test";

export class SignInPage extends DefaultLayout {
  readonly path: string = "/sign-in";

  get emailInput(): Locator {
    return this.page.getByTestId("email-input");
  }

  get passwordInput(): Locator {
    return this.page.getByTestId("password-input");
  }

  get signInButton(): Locator {
    return this.page.getByTestId("login-button");
  }

  get signInErrorAlert(): Locator {
    return this.page.getByTestId("sign-in-error-alert");
  }

  async signIn(email: string, password: string): Promise<void> {
    await this.emailInput.fill(email);
    await this.passwordInput.fill(password);
    await this.signInButton.click();
  }
}
